### PR TITLE
(QA-2162) Add PowerShell Helpers

### PR DESCRIPTION
**Pull request by cowofevil**
Add methods for executing PowerShell commands on a host.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require "bundler/gem_tasks"
+
+require 'rspec/core/rake_task'
+
+task :default => :test
```
Comment (zreichert) [inline Rakefile]: If you are going to have more than one rake task you should probably not point default at test.  
If you want to have a default task this is a nice option.

```
task :default do
  system "rake --tasks"
end
```

This should be the equivalent of 

```
rake -T 
```

File: lib/beaker_windows/powershell.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+      # Wrap the command in quotes when not encoded
+      command = "\"#{command}\"" unless opts[:EncodedCommand]
+
+      # Coerce all keys to be strings.
+      opts.each do |k, v|
+        ps_opts[k.to_s] = v unless (k == :verify_cmd) || (k == :excep_fail)
+      end
+
+      return powershell(command, ps_opts)
```
Comment (zreichert) [inline lib/beaker_windows/powershell.rb:61]:
```
powershell()
```

Is a beaker method, correct?

Comment (cowofevil) [inline lib/beaker_windows/powershell.rb:61]: Yes.

File: spec/beaker_windows/powershell_spec.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    it 'should validate command and wrap in try/catch when specified' do
+      exp = "-Command \"try { if ( #{ps_cmd} ) { exit 0 } else { exit 1 } } " \
+            "catch { Write-Host \\$_.Exception.Message; exit 1 }\""
+
+      expect(dummy_class.exec_ps_cmd(ps_cmd, :excep_fail => true, :verify_cmd => true).args.last).to eq(exp)
+    end
+
+    it 'should encode a command if specified' do
+      exp = "-EncodedCommand dAByAHkAIAB7ACAAVwByAGkAdABlAC0ASABvAHMAdAAgACcAS" \
```
Comment (zreichert) [inline spec/beaker_windows/powershell_spec.rb:52]: I thought you preferred?

```
exp = <<-EOS
...
EOS
```

Comment (cowofevil) [inline spec/beaker_windows/powershell_spec.rb:52]: This is actually a single string that can't be broken up into a here-doc.

**Pull request merged**